In [1]:
#Numpy one-d arrays
import pandas as pd
import numpy as np
import scipy.stats as stats

In [2]:
data=np.array([6,7,7,12,13,13,15,15,16,19,22])

In [3]:
stats.zscore(data)

array([-1.48977072, -1.28233429, -1.28233429, -0.24515214, -0.03771571,
       -0.03771571,  0.37715714,  0.37715714,  0.58459357,  1.20690286,
        1.82921214])

Each zscore tells us how many std.devs away an individual value if from the mean.

For eg: The first value of '6' in the array is 1.489 std deviations below the mean.
The seventh value of '15' in the array is 0.377 std deviations above the mean.

In [4]:
# numpy multi dimensional arrays
data=np.array([[5,6,7,7,8],[8,8,8,9,9],[2,2,4,4,5]])
#zcore
stats.zscore(data,axis=1)

array([[-1.56892908, -0.58834841,  0.39223227,  0.39223227,  1.37281295],
       [-0.81649658, -0.81649658, -0.81649658,  1.22474487,  1.22474487],
       [-1.16666667, -1.16666667,  0.5       ,  0.5       ,  1.33333333]])

The first value of '5' in the first array is 1.568 std dev below the mean of its array.
The first value of '8' in the second array is 0.816 std dev below the mean
Thf third value of '4' in the third array is 0.5 std dev above the mean

In [5]:
#Pandas df
data=pd.DataFrame(np.random.randint(0,10,size=(5,3)),columns=['A','B','C'])
data

A  B  C
0  8  6  2
1  2  9  4
2  1  0  9
3  2  5  5
4  3  0  0

In [6]:
data.apply(stats.zscore)

A         B         C
0  1.933975  0.567962 -0.659380
1 -0.483494  1.419905  0.000000
2 -0.886405 -1.135924  1.648451
3 -0.483494  0.283981  0.329690
4 -0.080582 -1.135924 -1.318761

In [7]:
bfsales=pd.read_csv('BlackFridaySales.csv')
bfsales.head()

User_ID Product_ID Gender   Age  Occupation City_Category  \
0  1000001  P00069042      F  0-17          10             A   
1  1000001  P00248942      F  0-17          10             A   
2  1000001  P00087842      F  0-17          10             A   
3  1000001  P00085442      F  0-17          10             A   
4  1000002  P00285442      M   55+          16             C   

  Stay_In_Current_City_Years  Marital_Status  Product_Category_1  \
0                          2               0                   3   
1                          2               0                   1   
2                          2               0                  12   
3                          2               0                  12   
4                         4+               0                   8   

   Product_Category_2  Product_Category_3  Purchase  
0                 NaN                 NaN      8370  
1                 6.0                14.0     15200  
2                 NaN                 NaN      1422  
3                14.0                 NaN      1057  
4                 NaN                 NaN      7969

In [9]:
stats.zscore(bfsales['Purchase'])

0        -0.177973
1         1.181756
2        -1.561193
3        -1.633858
4        -0.257805
            ...   
550063   -1.771025
550064   -1.770428
550065   -1.817013
550066   -1.771623
550067   -1.746737
Name: Purchase, Length: 550068, dtype: float64